pkg/kubelet/kubelet_node_status.go

```diff
@@ -273,6 +273,23 @@ func (kl *Kubelet) initialNode() (*v1.Node, error) {
 			glog.Infof("Adding node label from cloud provider: %s=%s", metav1.LabelInstanceType, instanceType)
 			node.ObjectMeta.Labels[metav1.LabelInstanceType] = instanceType
 		}
+
+		labels, err := instances.NodeLabels(kl.nodeName)
+		if err != nil {
+			return nil, err
+		}
+		if labels != nil {
+			prefix = "cloud.alpha.kubernetes.io/"
+			for name, value := range labels {
+				key := prefix + name
+				if len(key) > 63 {
+					glog.Warningf("Key: %s is too long, skipping", key)
```

Comment:
given the length of the prefix, you have 37 chars to work with.

on aws, the maximum key length is 127 characters.  it also looks like aws allows a broader character set.  a poorly written cloud provider that did not validate the labels or do character conversion could cause an unintentional kubelet to not be able to report for duty.  it would be nice if we can protect for those scenarios centrally.